### PR TITLE
soc: arm: stm32l4 serie low power modes substates

### PR DESCRIPTION
**Pull request by FRASTM**
Change the substate-id range from 1-3 as described in the include/power/power_state.h
this is changing the range in pm_power_state_set() of the soc/arm/st_stm32/stm32l4/power.c
The debug config will let the clocks active in STOP mode at init.The debug config will let the clocks active in STOP mode at init.
Following the #33131, the power.c part of the stm32wb is modified in the same way.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: soc/arm/st_stm32/stm32l4/power.c [hunk 1]
```diff
@@ -121,16 +105,15 @@ void pm_power_state_exit_post_ops(struct pm_state_info info)
...
 	ARG_UNUSED(dev);
 
-	ret = irq_lock();
-
 	/* enable Power clock */
 	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR);
 
-	irq_unlock(ret);
+#ifdef CONFIG_DEBUG
+		/* Enable the Debug Module during STOP mode */
```
Comment (erwango) [inline soc/arm/st_stm32/stm32l4/power.c:114]: Please fix indentation

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32l4/power.c (hunk 1 above)
Comment (FRASTM) [reply soc/arm/st_stm32/stm32l4/power.c:114]: done

**Review by erwango — APPROVED** [review]

**Comment by FRASTM** [comment]
plus a comment in the commit msg for the minor fix in the Kconfig

**Comment by FRASTM** [comment]
rebase and change the substate range

**Review by ABOSTM — CHANGES_REQUESTED** [review]
Commit Message:
```
The low power states 0∕1∕2 are added to the stm32l476rg
nucleo board, with (default min residency values).
```
Sentence not finished: With what ? (or wrong parenthesis)
Otherwise LGTM

**Comment by FRASTM** [comment]
rebase
changed info.substate_id in the pm_power_state_exit_post_ops function

**Review by ABOSTM — APPROVED** [review]
LGTM

**Pull request merged**